system/src/Grav/Common/Media/Traits/VideoMediaTrait.php

```diff
@@ -51,14 +54,27 @@ public function playsinline($status = false)
     /**
      * Parsedown element for source display mode
      *
-     * @param  array $attributes
-     * @param  bool $reset
+     * @param array $attributes
+     * @param bool $reset
      * @return array
      */
     protected function sourceParsedownElement(array $attributes, $reset = true)
     {
         $location = $this->url($reset);
 
+        if (!isset($attributes['poster']) || ($attributes['poster'] !== 0 && $attributes['poster'] !== '0')) {
```

Comment:
It might be simplified in this way:
```suggestion
        $this->setPoster($attributes);
        $this->removePoster($attributes);
```

To make it work, you need to create the following methods in the `VideoMediaTrait` trait:
```php
    private function isPosterUnknown(array $attributes): bool
    {
        return !isset($attributes['poster']) || (int) $attributes['poster'] !== 0;
    }

    private function removePoster(array $attributes): void
    {
        if ($this->isPosterUnknown($attributes)) {
            return;
        }

        unset($attributes['poster']);
    }

    private function setPoster(array $attributes): void
    {
        if ($this->isPosterUnknown($attributes) && $this->thumbnailExists('page')) {
            $thumb = $this->get('thumbnails.page', false);

            if ($thumb) {
                $thumb = $thumb instanceof ThumbnailImageMedium
                    ? $thumb
                    : MediumFactory::fromFile($thumb, ['type' => 'thumbnail']);

                $attributes['poster'] = $thumb->url();
            }
        }
    }
```

The names of new methods may be different. I want to show you the idea. Of course, you'll need to double-check that and ensure it works as expected.